### ISSUE TITLE
Rubocop ignores Exclude in inherited configuration file.

### ISSUE THREAD

**Issue by JoeCohen** [rubocop/rubocop#13091]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

RuboCop should respect all Exclude's in an inherited configuration file.

## Actual behavior

RuboCop reports violations of a cop for a file listed in `.rubocop_todo.yml` under `Exclude` for that cop.

`.rubocop.yml` includes:
```ruby
inherit_from: .rubocop_todo.yml
```
`.rubocop_todo.yml` includes a very long list of files to Exclude for `Metrics/AbcSize`
```ruby
# Offense count: 350
# Configuration parameters: AllowedMethods, AllowedPatterns, CountRepeatedAttributes, Max.
Metrics/AbcSize:
  Exclude:
    - 'app/classes/api2/core/parameters.rb'
    - 'app/classes/api2/external_link_api.rb'
    - 'app/classes/api2/image_api.rb'
    - 'app/classes/api2/location_api.rb'
    - 'app/classes/api2/name_api.rb'
    - 'app/classes/api2/observation_api.rb'
    - 'app/classes/api2/parsers/time_parser.rb'
    - 'app/classes/api2/sequence_api.rb'
    - 'app/classes/api2/species_list_api.rb'
    - 'app/classes/api2/user_api.rb'
    - 'app/classes/auto_complete/by_word.rb'
    - 'app/classes/auto_complete/for_clade.rb'
    - 'app/classes/auto_complete/for_herbarium.rb'
    - 'app/classes/auto_complete/for_location.rb'
    - 'app/classes/auto_complete/for_location_containing.rb'
    - 'app/classes/auto_complete/for_name.rb'
    - 'app/classes/auto_complete/for_region.rb'
    - 'app/classes/auto_complete/for_user.rb'
    - 'app/classes/checklist.rb'
    - 'app/classes/eol_data.rb'
    - 'app/classes/ip_stats.rb'
    - 'app/classes/mappable/collapsible_collection_of_objects.rb'
    - 'app/classes/mappable/map_set.rb'
    - 'app/classes/pattern_search/base.rb'
    - 'app/classes/pattern_search/error/bad_term_error.rb'
    - 'app/classes/pattern_search/term.rb'
    - 'app/classes/pattern_search/term/dates.rb'
    - 'app/classes/prawn/field_slip_card.rb'
    - 'app/classes/project_lat_lngs.rb'
    - 'app/classes/query/herbarium_base.rb'
    - 'app/classes/query/modules/conditions.rb'
    - 'app/classes/query/modules/datetime.rb'
    - 'app/classes/query/modules/high_level_queries.rb'
    - 'app/classes/query/modules/joining.rb'
    - 'app/classes/query/modules/sequence_operators.rb'
    - 'app/classes/query/modules/sql.rb'
    - 'app/classes/query/modules/validation.rb'
    - 'app/classes/report/adolf.rb'
    - 'app/classes/report/darwin/observations.rb'
    - 'app/classes/report/fundis.rb'
    - 'app/classes/report/raw.rb'
    - 'app/classes/report/symbiota.rb'
    - 'app/classes/site_data.rb'
    - 'app/classes/suggestion.rb'
    - 'app/classes/tree.rb'
    - 'app/controllers/account/login_controller.rb'
    - 'app/controllers/account/profile/images_controller.rb'
    - 'app/controllers/account/profile_controller.rb'
    - 'app/controllers/account_controller.rb'
    - 'app/controllers/admin/blocked_ips_controller.rb'
    - 'app/controllers/admin/session_controller.rb'
    - 'app/controllers/api2_controller.rb'
    - 'app/controllers/application_controller.rb'
    - 'app/controllers/articles_controller.rb'
    - 'app/controllers/checklists_controller.rb'
    - 'app/controllers/collection_numbers_controller.rb'
    - 'app/controllers/comments_controller.rb'
    - 'app/controllers/concerns/descriptions.rb'
    - 'app/controllers/concerns/descriptions/merges.rb'
    - 'app/controllers/concerns/descriptions/moves.rb'
    - 'app/controllers/concerns/descriptions/permissions.rb'
    - 'app/controllers/concerns/descriptions/publish.rb'
    - 'app/controllers/field_slips_controller.rb'
    - 'app/controllers/herbaria/curator_requests_controller.rb'
    - 'app/controllers/herbaria/curators_controller.rb'
    - 'app/controllers/herbaria_controller.rb'
    - 'app/controllers/herbarium_records_controller.rb'
    - 'app/controllers/images/votes/anonymity_controller.rb'
    - 'app/controllers/images/votes_controller.rb'
    - 'app/controllers/interests_controller.rb'
    - 'app/controllers/locations/descriptions_controller.rb'
    - 'app/controllers/locations_controller.rb'
    - 'app/controllers/names/classification/inherit_controller.rb'
    - 'app/controllers/names/descriptions_controller.rb'
    - 'app/controllers/names/eol_data/preview_controller.rb'
    - 'app/controllers/names/synonyms/deprecate_controller.rb'
    - 'app/controllers/names/synonyms_controller.rb'
    - 'app/controllers/names/trackers_controller.rb'
    - 'app/controllers/names_controller.rb'
    - 'app/controllers/observations/images_controller.rb'
    - 'app/controllers/observations/locations_controller.rb'
    - 'app/controllers/observations/maps_controller.rb'
    - 'app/controllers/observations/namings/votes_controller.rb'
    - 'app/controllers/observations/namings_controller.rb'
    - 'app/controllers/observations/species_lists_controller.rb'
    - 'app/controllers/observations_controller/create.rb'
    - 'app/controllers/observations_controller/destroy.rb'
    - 'app/controllers/observations_controller/edit_and_update.rb'
    - 'app/controllers/observations_controller/new.rb'
    - 'app/controllers/observations_controller/shared_form_methods.rb'
    - 'app/controllers/observations_controller/show.rb'
    - 'app/controllers/observations_controller/validators.rb'
    - 'app/controllers/projects/admin_requests_controller.rb'
    - 'app/controllers/projects/field_slips_controller.rb'
    - 'app/controllers/projects/members_controller.rb'
    - 'app/controllers/projects_controller.rb'
    - 'app/controllers/publications_controller.rb'
    - 'app/controllers/rss_logs_controller.rb'
    - 'app/controllers/species_lists/name_lists_controller.rb'
    - 'app/controllers/species_lists/projects_controller.rb'
    - 'app/controllers/species_lists/shared_private_methods.rb'
    - 'app/controllers/species_lists/shared_render_methods.rb'
    - 'app/controllers/translations_controller.rb'
    - 'app/extensions/string_extensions.rb'
    - 'app/extensions/symbol_extensions.rb'
    - 'app/helpers/api2_inline_helper.rb'
    - 'app/helpers/api_keys_helper.rb'
    - 'app/helpers/application_helper.rb'
    - 'app/helpers/carousel_helper.rb'
    - 'app/helpers/content_helper.rb'
    - 'app/helpers/descriptions_helper.rb'
    - 'app/helpers/footer_helper.rb'
    - 'app/helpers/forms_helper.rb'
    - 'app/helpers/images_helper.rb'
    - 'app/helpers/link_helper.rb'
    - 'app/helpers/map_helper.rb'
    - 'app/helpers/matrix_box_helper.rb'
    - 'app/helpers/names_helper.rb'
    - 'app/helpers/namings_helper.rb'
    - 'app/helpers/observations_helper.rb'
    - 'app/helpers/pagination_helper.rb'
    - 'app/helpers/table_helper.rb'
    - 'app/helpers/tabs/descriptions_helper.rb'
    - 'app/helpers/tabs/projects_helper.rb'
    - 'app/helpers/title_and_tabset_helper.rb'
    - 'app/helpers/user_stats_helper.rb'
    - 'app/jobs/field_slip_job.rb'
    - 'app/models/abstract_model.rb'
    - 'app/models/api_key.rb'
    - 'app/models/description.rb'
    - 'app/models/field_slip.rb'
    - 'app/models/image.rb'
    - 'app/models/interest.rb'
    - 'app/models/language_exporter.rb'
    - 'app/models/location.rb'
    - 'app/models/location_description.rb'
    - 'app/models/name.rb'
    - 'app/models/name/change.rb'
    - 'app/models/name/create.rb'
    - 'app/models/name/format.rb'
    - 'app/models/name/lifeform.rb'
    - 'app/models/name/parse.rb'
    - 'app/models/name/spelling.rb'
    - 'app/models/name/taxonomy.rb'
    - 'app/models/name_description.rb'
    - 'app/models/name_parse.rb'
    - 'app/models/name_sorter.rb'
    - 'app/models/name_tracker.rb'
    - 'app/models/naming.rb'
    - 'app/models/naming/name_resolver.rb'
    - 'app/models/observation.rb'
    - 'app/models/observation/consensus_calculator.rb'
    - 'app/models/observation/naming_consensus.rb'
    - 'app/models/project.rb'
    - 'app/models/queued_email.rb'
    - 'app/models/queued_email/location_change.rb'
    - 'app/models/queued_email/name_change.rb'
    - 'app/models/queued_email/observation_change.rb'
    - 'app/models/species_list.rb'
    - 'app/models/synonym.rb'
    - 'app/models/translation_string.rb'
    - 'app/models/user.rb'
    - 'app/models/user_stats.rb'
    - 'app/models/visual_group.rb'
    - 'app/models/vote.rb'
    - 'app/presenters/image_presenter.rb'
    - 'app/presenters/matrix_box_presenter.rb'
    - 'lib/core_extensions/action_view/log_subscriber.rb'
    - 'lib/tasks/lang.rake'
    - 'script/bulk_name_change'
    - 'script/config.rb'
    - 'script/refresh_sitemap'
    - 'script/update_ip_stats.rb'
```

## Steps to reproduce the problem

- `git clone https://github.com/MushroomObserver/mushroom-observer`
- `cd mushroom-observer`  
- `rubocop -f simple app/helpers/forms_helper.rb --only Metrics/AbcSize`

Will yield this output:
```ruby
== app/helpers/forms_helper.rb ==
C: 12:  3: Metrics/AbcSize: Assignment Branch Condition size for submit_button is too high. [<7, 20, 6> 22.02/17]
C: 87:  3: Metrics/AbcSize: Assignment Branch Condition size for check_box_with_label is too high. [<4, 25, 2> 25.4/17]
C:121:  3: Metrics/AbcSize: Assignment Branch Condition size for radio_with_label is too high. [<4, 27, 2> 27.37/17]
C:155:  3: Metrics/AbcSize: Assignment Branch Condition size for text_field_with_label is too high. [<9, 72, 8> 73/17]
C:332:  3: Metrics/AbcSize: Assignment Branch Condition size for text_area_with_label is too high. [<9, 30, 4> 31.58/17]
C:354:  3: Metrics/AbcSize: Assignment Branch Condition size for select_with_label is too high. [<9, 31, 4> 32.53/17]
C:395:  3: Metrics/AbcSize: Assignment Branch Condition size for date_select_with_label is too high. [<11, 41, 6> 42.87/17]
C:419:  3: Metrics/AbcSize: Assignment Branch Condition size for date_select_opts is too high. [<8, 22, 13> 26.78/17]
C:436:  3: Metrics/AbcSize: Assignment Branch Condition size for number_field_with_label is too high. [<6, 17, 1> 18.06/17]
C:482:  3: Metrics/AbcSize: Assignment Branch Condition size for hidden_field_with_label is too high. [<3, 18, 2> 18.36/17]
C:526:  3: Metrics/AbcSize: Assignment Branch Condition size for file_field_with_label is too high. [<8, 34, 1> 34.94/17]
C:644:  3: Metrics/AbcSize: Assignment Branch Condition size for check_for_help_block is too high. [<3, 23, 3> 23.39/17]

1 file inspected, 12 offenses detected
```

## RuboCop version

```ruby
% rubocop -V -f simple app/helpers/forms_helper.rb --only Metrics/AbcSize
1.65.1 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.3.0) [x86_64-darwin23]
  - rubocop-performance 1.21.1
  - rubocop-rails 2.25.1
```

Other: MacOS Sonoma 14.5, % ruby --version
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x86_64-darwin23]

**Comment by Earlopain** [comment]
Yes, I find this behaviour rather unfortunate. There's a related issue here https://github.com/rubocop/rubocop/issues/9325 and this does affect the generated todo file as well.

The solution is to make use of the `inherit_mode` option, the other issue contains a snippet with that.

**Comment by JoeCohen** [comment]
@Earlopain: Thanks for the pointer to the fix/workaround. It's a big improvement.
There is one remaining or new issue. The fix is not working for the `Metrics/ModuleLength` cop.  
(As a result, I disabled that cop in one file. This, in turn, caused an autoconfig error: the todo file does not include a `Metrics/ModuleLength` for that one file.)

**Comment by Earlopain** [comment]
Is it possible that you generated the todo while the disable comment was still present? I see that the application with this error is open source and while trying it out myself it just worked, the `forms_helper` got added to the todo file just fine.

**Comment by JoeCohen** [comment]
@Earlopain Thanks for the response.
My mistake. RuboCop is behaving correctly on its own.
(What I was seeing was an issue in VSCode, which ignored the `Metrics/ModuleLength` Exclude.)
